File: ExploringActivityTemplates/app/src/main/java/njit/mt/exploringactivitytemplates/MainActivity.java
package njit.mt.exploringactivitytemplates;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import njit.mt.exploringactivitytemplates.ui.main.MainFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.container, MainFragment.newInstance())
                    .commitNow();
        }
    }
}

File: Project/ChaoticBattleShip/app/src/main/java/com/ethereallab/chaoticbattleship/MainActivity.kt
package com.ethereallab.chaoticbattleship

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.onNavDestinationSelected
import androidx.navigation.ui.setupWithNavController
import com.ethereallab.chaoticbattleship.databinding.ActivityMainBinding
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {

    private lateinit var auth: FirebaseAuth
    private lateinit var binding: ActivityMainBinding
    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Initialize View Binding
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Setup Navigation
        val navHostFragment = supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
        navController = navHostFragment.navController

        // Link BottomNavigationView with NavController
        binding.bottomNavigation.setOnItemSelectedListener { item ->
            when (item.itemId) {
                R.id.lobbyFragment -> {
                    navController.navigate(R.id.lobbyFragment) // Explicitly navigate to LobbyFragment
                    true
                }
                else -> item.onNavDestinationSelected(navController) || super.onOptionsItemSelected(item)
            }
        }
    }
}
